specs/ServerCertificate.md

```diff
@@ -0,0 +1,299 @@
+             Callback<ICoreWebView2ReceivingServerCertificateErrorEventHandler>(
+             [this](
+               ICoreWebView2* sender,
+               ICoreWebView2ReceivingServerCertificateErrorEventArgs* args) {
+               COREWEBVIEW2_WEB_ERROR_STATUS errorStatus;
+               CHECK_FAILURE(args->get_ErrorStatus(&errorStatus));
+
+               wil::com_ptr<ICoreWebView2Certificate> certificate = nullptr;
+               CHECK_FAILURE(args->get_ServerCertificate(&certificate));
+
+               // Continues with the navigation to a server with a TLS certificate if
+               // the error status is of type `COREWEBVIEW2_WEB_ERROR_STATUS_CERTIFICATE_IS_INVALID`
+               // and trusted by the host app.
+               if (errorStatus == COREWEBVIEW2_WEB_ERROR_STATUS_CERTIFICATE_IS_INVALID &&
+                            ValidateServerCertificate(certificate.get()))
+               {
+                 CHECK_FAILURE(args->put_Handled(TRUE));
+               }
+               else
+               {
+                 // Cancel the request for other TLS certificate error types or if untrusted by the host app.
+                 CHECK_FAILURE(args->put_Cancel(TRUE));
+               }
+               return S_OK;
+             })
+             .Get(),
+         &m_ServerCertificateErrorToken));
+    }
+    else
+    {
+      CHECK_FAILURE(m_webview->remove_ReceivingServerCertificateError(
+        m_ServerCertificateErrorToken));
+      m_ServerCertificateErrorToken.value = 0;
+    }
+  }
+  else
+  {
+    FeatureNotAvailable();
+  }
+}
+
+// Function to validate the server certificate for untrusted root or self-signed certificate.
+bool ValidateServerCertificate(ICoreWebView2Certificate* certificate)
+{
+    // You may want to validate certificates in different ways depending on your app and scenario.
+    // One way might be the following:
+    // First, get the list of host app trusted certificates and its thumbprint.
+
+    // Then get the last chain element using `ICoreWebView2Certificate::get_PemEncodedIssuerCertificateChain`
+    // that contains the raw data of the untrusted root CA/self-signed certificate. Get the untrusted
+    // root CA/self signed certificate thumbprint from the raw certificate data and
+    // validate the thumbprint against the host app trusted certificate list.
+
+    // Finally, return true if it exists in the host app's certificate trusted list, or otherwise return false.
+    return true;
+}
+
+// This example clears the TLS decision in response to proceeding with TLS certificate errors.
+if (m_webview)
+{
+  CHECK_FAILURE(m_webView->ClearServerCertificateErrorOverrideCache(
+        Callback<
+            ICoreWebView2ClearServerCertificateErrorOverrideCacheCompletedHandler>(
+            [](HRESULT error, BOOL isSuccessful) -> HRESULT {
+                CHECK_FAILURE(error);
+                auto showDialog = [isSuccessful] {
```

Comment:
I had similar question above. We can represent exceptional failure in the HRESULT error parameter in COM or by throwing an exception in .NET/WinRT and we don't have non-exceptional failure. So we don't need a BOOL parameter. This can just be (HRESULT error) in COM and an IAsyncAction in WinRT.